File: app/src/main/java/com/example/cbanda/achieve/Models/Repository/GoalRepositoryImpl.java
package com.example.cbanda.achieve.Models.Repository;

import android.arch.lifecycle.LiveData;

import com.example.cbanda.achieve.Models.db.GoalDao;
import com.example.cbanda.achieve.Models.db.Goal;

import java.util.List;

/**
 * Created by CBanda on 2018/02/07.
 */

public class GoalRepositoryImpl implements GoalRepository  {

private GoalDao goalDao;

public GoalRepositoryImpl(GoalDao goalDao)
{
    this.goalDao=goalDao;
}
    @Override
    public LiveData<List<Goal>> getListGoals() {
        return goalDao.getListGoals();
    }

    @Override
    public void addGoal(Goal goal) {
    goalDao.addGoal(goal);
    }
}

File: app/src/main/java/com/example/cbanda/achieve/models/repository/GoalRepositoryImpl.kt
package com.example.cbanda.achieve.models.repository

import androidx.lifecycle.LiveData

import com.example.cbanda.achieve.models.db.Goal
import com.example.cbanda.achieve.models.db.GoalDao

class GoalRepositoryImpl(private val goalDao: GoalDao) : GoalRepository {

    override val listGoals: LiveData<List<Goal>>
        get() = goalDao.listGoals

    override fun getGoalById(id: Int): LiveData<Goal> {
        return goalDao.getGoalById(id)
    }


    override fun addGoal(goal: Goal) {
        InsertAsyncTask(goalDao).execute(goal)
    }

}
